vividus/src/main/resources/org/vividus/spring-configuration.xml

```diff
@@ -3,7 +3,11 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://www.springframework.org/schema/beans https://www.springframework.org/schema/beans/spring-beans.xsd">
 
-    <bean id="properties" class="org.vividus.configuration.PropertiesDecryptor" factory-method="decryptProperties">
+    <bean id="properties" class="org.vividus.configuration.SystemPropertiesExtractor" factory-method="process">
```

Comment:
system properties should be set before bean factory is opened, otherwise some beans may be initialized with default system properties and race conditions are possible here